httpcore/backends/sync.py

```diff
@@ -16,6 +16,47 @@
 from .._utils import is_socket_readable
 from .base import NetworkBackend, NetworkStream
 
+SOCKET_OPTIONS = typing.Union[
+    typing.Tuple[int, int, int],
+    typing.Tuple[int, int, typing.Union[bytes, bytearray]],
+    typing.Tuple[int, int, None, int],
+]
+
+
+def create_connection(
```

Comment:
This function is very similar to the built-in `create_connection` function, but it can also accept socket options.
Because there are options that affect the connection itself, we must set socket options before connecting; we cannot set options after connecting.

Reply:
Okay, I'd suggest we add a comment noting that.

Perhaps rephrased to "This function is equivalent to Python's `socket.create_connection()` function, but also accepts socket options." and linking to the source on GitHub?